src/Debug/Pretty/Simple.hs

```diff
@@ -228,7 +246,83 @@ pTraceMarker = traceMarker . unpack . pString
 --
 -- @since 2.0.1.0
 pTraceMarkerIO :: String -> IO ()
-pTraceMarkerIO = traceMarkerIO . unpack . pString
+pTraceMarkerIO = traceMarkerIO . unpack <=< pStringTTYIO
+
+------------------------------------------
+-- Helpers
+------------------------------------------
+
+pStringTTYIO :: String -> IO Text
+pStringTTYIO v = checkTTY defaultOutputOptionsDarkBg <&> \o -> pStringOpt o v
+
+pStringTTY :: String -> Text
+pStringTTY = unsafePerformIO . pStringTTYIO
```

Comment:
As I mentioned in the commit message, `unsafePerformIO` seems unavoidable.

Reply:
It looks like the `trace` functions are using `unsafePerformIO` internally, so this seems reasonable:

http://hackage.haskell.org/package/base-4.12.0.0/docs/src/Debug.Trace.html#trace